docs(datasource): Add additional examples for custom datasource (#23558)
renovatebot/renovate: lib/modules/datasource/custom/readme.md

  },
}
```

### Hashicorp

You can use this configuration to request the newest versions of the hashicorp products:

```json
{
  "regexManagers": [
    {
      "fileMatch": ["\\.yml$"],
      "datasourceTemplate": "custom.hashicorp",
      "matchStrings": [
        "#\\s*renovate:\\s*(datasource=(?<datasource>.*?) )?depName=(?<depName>.*?)( versioning=(?<versioning>.*?))?\\s*\\w*:\\s*(?<currentValue>.*)\\s"
      ],
      "versioningTemplate": "{{#if versioning}}{{{versioning}}}{{else}}semver{{/if}}"
    }
  ],
  "customDatasources": {
    "hashicorp": {
      "defaultRegistryUrlTemplate": "https://api.releases.hashicorp.com/v1/releases/{{packageName}}?license_class=oss",
      "transformTemplates": [
        "{ \"releases\": $map($, function($v) { { \"version\": $v.version, \"releaseTimestamp\": $v.timestamp_created, \"changelogUrl\": $v.url_changelog, \"sourceUrl\": $v.url_source_repository } }), \"homepage\": $[0].url_project_website, \"sourceUrl\": $[0].url_source_repository }"
      ]
    }
  },
}
```

> E.g. to have the latest Nomad version in your ansible variables, use this snippet after adding the above configuration:

Reviewer: ```suggestion
To have the latest Nomad version in your Ansible variables, use this snippet _after_ adding the above configuration:
```

```yaml
# renovate: depName=nomad
nomad_version: 1.6.0
```

> ### Custom Offline Dependencies

Reviewer: ```suggestion
### Custom offline dependencies
```

> Sometimes the source of the dependency versions is not available via an API but has to be generated manually. For this purpose, you can manually create dependency "files" (similar to an API) served via http(s) for renovate to access. For example, imagine the following file `versiontracker.json` for the software `something``:

Reviewer: ```suggestion
Sometimes the "dependency version source" is _not_ available via an API.
To work around a missing API, you can create dependency "files". These files are served via HTTP(S), so that Renovate can access them.
For example, imagine the following file `versiontracker.json` for the software `something``:
```

```json
[
    {
        "version": "77"
    },
    {
        "version": "76"
    },
]
```

> Which can be ingested by renovate using the following custom datasource (using nexus as a webserver in this case):

Reviewer: ```suggestion
By writing a custom datasource, Renovate can process the `versiontracker.json` file, see below.
This example uses Nexus as the webserver.
```

```json
{
  "customDatasources": {
    "nexus_generic": {
      "defaultRegistryUrlTemplate": "https://nexus.example.com/repository/versiontrackers/{{packageName}}/versiontracker.json",
      "transformTemplates": [
        "{ \"releases\": $map($, function($v) { { \"version\": $v.version, \"sourceUrl\": $v.filelink } }) }"
      ]
    }
  },
}
```

> This could be used to update ansible yaml files with the latest version through a regex manager, e.g. with the following ansible content:

Reviewer: ```suggestion
This could be used to update Ansible YAML files with the latest version through a regex manager.
For example, with the following Ansible content:
```
